public/src/Helpers.ts

```diff
@@ -135,12 +135,12 @@ export const getPassedTestsCount = (score: Score[]): string => {
     return `${passedTests}/${totalTests}`
 }
 
-export const isValid = (elm: User | EnrollmentLink): boolean => {
+export const isValid = (elm: User | Enrollment): boolean => {
     if (elm instanceof User) {
         return elm.Name.length > 0 && elm.Email.length > 0 && elm.StudentID.length > 0
     }
-    if (elm instanceof EnrollmentLink) {
-        return elm.enrollment?.user !== undefined && elm.submissions.length > 0
+    if (elm instanceof Enrollment) {
+        return elm.user !== undefined && isValid(elm.user)
```

Comment:
I recall that deep source didn't like comparing with `undefined`, but didn't check. Maybe it should be changed to `null` instead?

Reply:
Apparently this helper is only used in tests in `isValid.test.tsx`. I removed it altogether and updated the test to use `state.isValid` that is derived from `state.self`